### DISCUSSION TITLE
how about rename rule luarocks.module to lua.module

### DISCUSSION THREAD

**Discussion by Freed-Wu** [xmake-io/xmake#5950] [Ideas]
- It just compile a binary module of lua, not directly related to luarocks.
- In the future, we can have rule python.module, nodejs.module, zsh.module, [octave.module](https://docs.octave.org/v6.2.0/Oct_002dFiles.html), [matlab.module](https://docs.octave.org/v6.2.0/Mex_002dFiles.html) ... I think lua.module can be a more regular name.

Another similar rule name is `platform.linux.driver`. IMO, driver is one type of kernel module, maybe `platform.linux.module` is more regular.

For compatibility, can allow to use old rule name, however, encourage new rule name.

In conclusion:

- rename `luarocks.module` to `lua.module`
- rename `platform.linux.driver` to `platform.linux.module`
- in the future, use `*.module` as the names of the rules

**Comment by waruqi** [comment]
you can open a pr.
